idaes/core/util/tests/test_model_diagnostics.py

```diff
@@ -0,0 +1,249 @@
+'''
+
+# Need to update
+from idaes.core.util.model_diagnostics import *
+
+
+# Author: Alex Dowling
+
+# This was from 
+# @pytest.fixture()
+def problem1():
+    m = pyo.ConcreteModel()
+
+    m.I = pyo.Set(initialize=[i for i in range(5)])
+
+    m.x = pyo.Var(m.I,bounds=(-10,10),initialize=1.0)
+
+    m.con1 = pyo.Constraint(expr=m.x[0] + m.x[1] - m.x[3] >= 10)
+    m.con2 = pyo.Constraint(expr=m.x[0]*m.x[3] + m.x[1] >= 0)
+    m.con3 = pyo.Constraint(expr=m.x[4]*m.x[3] + m.x[0]*m.x[3] - m.x[4] == 0)
+
+    m.obj = pyo.Objective(expr=sum(m.x[i]**2 for i in m.I))
+
+    return m
+
+def example2(with_degenerate_constraint=True):
+    ''' Create the Pyomo model for Example 2
+    
+    Arguments:
+        with_degenerate_constraint: Boolean, if True, include the redundant linear constraint
+    
+    Returns:
+        m2: Pyomo model
+    '''
+    
+    m2 = pyo.ConcreteModel()
+
+    m2.I = pyo.Set(initialize=[i for i in range(1,4)])
+
+    m2.x = pyo.Var(m2.I,bounds=(0,5),initialize=1.0)
+
+    m2.con1 = pyo.Constraint(expr=m2.x[1] + m2.x[2] >= 1)
+    m2.con2 = pyo.Constraint(expr=m2.x[1] + m2.x[2] + m2.x[3] == 1)
+    m2.con3 = pyo.Constraint(expr=m2.x[2] - 2*m2.x[3] <= 1)
+    m2.con4 = pyo.Constraint(expr=m2.x[1] + m2.x[3] >= 1)
+    
+    if with_degenerate_constraint:
+        m2.con5 = pyo.Constraint(expr=m2.x[1] + m2.x[2] + m2.x[3] == 1)
+
+    m2.obj = pyo.Objective(expr=sum(m2.x[i] for i in m2.I))
+    
+    return m2
+
+def extract_constraint_names(cs):
+    ''' Get constraint names from ComponentSet
+    
+    Arguments:
+        cs: ComponentSet object
+        
+    Return:
+        constraint_names: list of constraint names (strings)
+            
+    '''
+    
+    constraint_names = []
+    for i in cs:
+        constraint_names.append(i.name)
+    return constraint_names
+
+# Problem 1
+@pytest.mark.unit
+def test_problem1():
+    # Create test problem
+    m = problem1()
+    
+    # Specify Ipopt as the solver
+    opt = pyo.SolverFactory('ipopt')
+
+    # Specifying an iteration limit of 0 allows us to inspect the initial point
+    opt.options['max_iter'] = 0
+
+    # "Solving" the model with an iteration limit of 0 load the initial point and applies
+    # any preprocessors (e.g., enforces bounds)
+    opt.solve(m, tee=True)
+
+    # Create Degeneracy Hunter object
+    dh = DegeneracyHunter(m) 
+    
+    # Find constraints with residuals > 0.1 
+    initial_point_constraints = dh.check_residuals(tol=0.1)
+    
+    # Check there are 2 constraints with large residuals
+    assert len(initial_point_constraints) == 2
+    
+    initial_point_constraint_names = extract_constraint_names(initial_point_constraints)
+    
+    # Check first constraint
+    assert initial_point_constraint_names[0] == 'con1'
+    
+    # Check second constraint
+    assert initial_point_constraint_names[1] == 'con3'
+    
+    # Specifying an iteration limit of 0 allows us to inspect the initial point
```

Comment:
Looks like this comment needs to be updated or removed

Reply:
Good catch.